### PR TITLE
Improve create-stack build time by using OCI layout

### PR DESCRIPTION
**Pull request by ryanmoran**
## Summary

The ihop.Client.Build method will now export the image from the Docker daemon as soon as the build completes, writing it to disk in OCI layout format. From this point on, the create-stack codebase will interact exclusively with the OCI layout image format. The advantage here is that we won't be shuttling an image back and forth between the Docker daemon and program memory or disk for every operation we need to perform. Instead, we can just manipulate the image on the local filesystem directly using the GGCR layout package.

## Use Cases

For large stacks like the Paketo Full stack, I was able to see a 10% improvement in build times on my local workstation. The vast majority of total time is taken up by the initial Docker build step which can't really be optimized externally. However, the remaining steps that add image labels and layers can. This change impacts those later pieces. For an image as large as the Full stack, we can still see significant improvements (on the order of minutes) from using an OCI layout as the concrete image format that we manipulate.

I haven't tested this in a GitHub runner, and would be interested to see how it fares in that environment, esp. for the Full stack.

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

File: internal/ihop/client.go [hunk 1]
```diff
@@ -52,27 +52,58 @@ type Image struct {
...
 
-	unbuffered bool
+	Actual v1.Image
+	Path   string
 }
 
-// ToDaemonImage returns the GGCR v1.Image associated with this Image.
-func (i Image) ToDaemonImage() (v1.Image, error) {
-	ref, err := name.ParseReference(i.Tag)
+func FromImage(img Image) (Image, error) {
```
Comment (sophiewigmore) [inline internal/ihop/client.go:59]: Is it possible to add a unit test for this function?

**Comment by sophiewigmore** [comment]
The code works locally for me, and in general this seems like a good idea! Besides the one comment I added about testing, the changes here seem good to me.

File: internal/ihop/client.go (hunk 1 above)
Comment (ryanmoran) [reply internal/ihop/client.go]: Yes. Thank you for the reminder. This is a place where I had planned to finish a refactoring and then never got to. I've finished that now, removing some fields and methods that are no longer needed.

**Review by sophiewigmore — APPROVED** [review]

File: scripts/unit.sh [hunk 2]
```diff
@@ -50,7 +50,7 @@ function unit::run() {
 
   testout=$(mktemp)
   pushd "${BUILDPACKDIR}" > /dev/null
-    if go test ./... -v -run "Unit|Example" | tee "${testout}"; then
+    if go test ./... -v -count=1 -run "Unit|Example" | tee "${testout}"; then
```
Comment (robdimsdale) [inline scripts/unit.sh:53]: Note for posterity - this is the idiomatic way to disable test caching in golang.

**Pull request merged**